datasource/gcp.go

```diff
@@ -125,3 +138,24 @@ func setEnvGoogleApplicationCredentials(values url.Values) error {
 	}
 	return nil
 }
+
+func getImpersonationTokenSource(ctx context.Context, values url.Values) (oauth2.TokenSource, error) {
+	impersonateServiceAccount := values.Get("impersonate_service_account")
+	if impersonateServiceAccount == "" {
+		return nil, nil
+	}
+	// Setting up options for service account impersonation
+	durationStr := values.Get("impersonate_service_account_duration")
+	if durationStr == "" {
+		durationStr = "300s"
+	}
+	duration, err := time.ParseDuration(durationStr)
```

Comment:
I suggest using https://github.com/k1LoW/duration to also accept more human readable values